perf(parser): Skip `try_parse` when current token is not an identifier

leaysgur opened the pull request:
Part of #11334

Check if it is worth doing `try_parse()` for the 8️⃣th hot path.

TSC: https://github.com/microsoft/TypeScript/blob/44d46714c81e0cadf9067cc32b649c63d2a4f436/src/compiler/parser.ts#L4912

Boshen commented:
### Merge activity

* **Jun 4, 8:42 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

leaysgur committed:
perf(parser): Skip `try_parse` when current token is not an identifier (#11475)

Part of #11334

Check if it is worth doing `try_parse()` for the 8️⃣th hot path.

TSC: https://github.com/microsoft/TypeScript/blob/44d46714c81e0cadf9067cc32b649c63d2a4f436/src/compiler/parser.ts#L4912
